api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java

```diff
@@ -27,149 +27,205 @@
+	 *             {@code com.sap.cloud.security.token} package.
 	 * @return user origin
 	 * @throws XSUserInfoException
 	 *             if attribute is not available in the authentication token
 	 */
+	@Deprecated
 	public String getOrigin() throws XSUserInfoException;
 
 	/**
 	 * Return identity zone
-	 *
+	 * 
+	 * @deprecated Can be replaced with
+	 *             {@code token.getClaimAsString(TokenClaims.XSUAA.SUBACCOUNT_ID)}
+	 *             from the {@code com.sap.cloud.security.token} package.
 	 * @return identity zone
 	 * @throws XSUserInfoException
 	 *             if attribute is not available in the authentication token
 	 */
+	@Deprecated
 	String getIdentityZone() throws XSUserInfoException;
 
 	/**
 	 * Return subaccount identifier
 	 * 
+	 * @deprecated Can be replaced with
+	 *             {@code token.getClaimAsString(TokenClaims.XSUAA.SUBACCOUNT_ID)}
+	 *             from the {@code com.sap.cloud.security.token} package.
 	 * @return subaccount identifier
 	 * @throws XSUserInfoException
 	 *             if attribute is not available in the authentication token
 	 */
+	@Deprecated
 	public String getSubaccountId() throws XSUserInfoException;
 
 	/**
 	 * Return the subdomain of this subaccount
 	 * 
+	 * @deprecated Can be replaced with
+	 *             {@code token.getClaimAsString(TokenClaims.XSUAA.SUBDOMAIN)} from
+	 *             the {@code com.sap.cloud.security.token} package.
 	 * @return subdomain
 	 * @throws XSUserInfoException
 	 *             if attribute is not available in the authentication token
 	 */
+	@Deprecated
 	public String getSubdomain() throws XSUserInfoException;
 
 	/**
 	 * Return the client id of the authentication token
-	 * 
+	 *
+	 * @deprecated Can be replaced with
+	 *             {@code token.getClaimAsString(TokenClaims.XSUAA.CLIENT_ID)} from
+	 *             the {@code com.sap.cloud.security.token} package.
 	 * @return client id
 	 * @throws XSUserInfoException
 	 *             if attribute is not available in the authentication token
 	 */
+	@Deprecated
 	public String getClientId() throws XSUserInfoException;
 
+	/**
+	 * @deprecated Can be replaced with {@code token.getClaimAsString(attribute)}
+	 *             from the {@code com.sap.cloud.security.token} package.
+	 */
 	@Deprecated
 	public String getJsonValue(String attribute) throws XSUserInfoException;
 
 	/**
 	 * Return the email of the user
 	 * 
+	 * @deprecated Can be replaced with
+	 *             {@code token.getClaimAsString(TokenClaims.XSUAA.EMAIL)} from the
+	 *             {@code com.sap.cloud.security.token} package.
+	 * 
 	 * @return email
 	 * @throws XSUserInfoException
 	 *             if attribute is not available in the authentication token
 	 */
+	@Deprecated
 	public String getEmail() throws XSUserInfoException;
 
 	@Deprecated // use getHdbToken
 	public String getDBToken() throws XSUserInfoException;
 
+	// TODO 14.01.20 c5295400: can this be replaced?
```

Comment:
No, have to check this with XSA 
but as it is also not supported in our Spring XsuaaToken...